gen/top_gen.ml

```diff
@@ -199,11 +196,15 @@ let get_fence n =
           f::fs,ns
       | _ -> [],all
 
-  let rec compile_proc pref chk loc_writes st p ro_prev init ns = match ns with
+  (* `ro_prev` if there is a previous load, `st` is the machine state
+     The callee from compile_proc
+                  Misc.identity false loc_writes A.st0 proc No      i    n *)
```

Comment:
I am not a big fun of comments like this. Either pref is always Misc.identity and we should get rid of the argument or the comment is wrong.

Reply:
Will rewrite something more descriptive.